sdks/java/extensions/protobuf/src/main/java/org/apache/beam/sdk/extensions/protobuf/ProtoSchemaLogicalTypes.java

```diff
@@ -31,8 +32,13 @@
   public abstract static class NanosType<T> implements LogicalType<T, Row> {
     private final String identifier;
 
-    private static final Schema SCHEMA =
-        Schema.builder().addInt64Field("seconds").addInt32Field("nanos").build();
+    private static final Schema SCHEMA;
+
+    static {
+      Schema schema = Schema.builder().addInt64Field("seconds").addInt32Field("nanos").build();
+      schema.setUUID(UUID.fromString("dc5915f1-6db8-3134-9444-4980fb088e92"));
```

Comment:
I don't think this is a great long-term solution. Do we need to get a fix in soon (like before the 2.19.0 cut) or is there time to consider a better solution? If we need to get it in soon I think we should file a follow-up jira and link it here.

Reply:
Well the other solution is making the schema not static. I do think we need to fix it for 2.19 otherwise the Proto support with Timestamp doesn't work on Dataflow.

Reply:
Could add a .buildNamed() to schema. This name could be used to use to generate a unique UUID from the name.
- Any better suggestion for buildNamed?
- Shouldn't we assert on setUUID with the UUID is already set?

Reply:
I think I like the (short-term) solution of just making it not static better. What do you think @reuvenlax?

Reply:
It's trickier then expected as the static **toRow** functions, is used by the bytebuddy generated code and you don't want to create a Schema object for each Row.

What about:

```
Schema SCHEMA = Schema.builder().addInt64Field("seconds").addInt32Field("nanos").buildNamed("NanosType");
```

This needs an extension to Schema.

Reply:
The more I think about it... the better the named schema's seem to be. You really want to have the schema static, as it's a know Schema for beam and deserves a unique UUID. I don't know if the **buildNamed** is a good method though, as named doesn't imply UUID and you don't necessarily can ask the name to the Schema again, as it's transformed to a UUID.